### ISSUE TITLE
Feature request: Store notes in one single file.

### ISSUE THREAD

**Issue by gujiaxi** [joostkremers/ebib#101]
In ebib, notes are stored in separate files according to citation key or something else determined by `ebib-notes-name-transform-function`. Could you consider such a feature for storing notes in one single file just like what [helm-bibtex](https://github.com/tmalsburg/helm-bibtex) does? Maybe it is a better choice if a variable can be provided for determining single-file or multi-file note-taking manner.

> Bibtex-completion supports two methods for storing notes. It can either store all notes in one file or store notes in multiple files, one file per publication.
> -- *Bibtex-completion, helm-bibtex, ivy-bibtex*

![](http://i.imgur.com/9xUhyjo.png)

**Comment by joostkremers** [comment]
Yes, I agree that would be useful. I'll look into it, but I can't promise it will be implemented very soon. I'll let you know if I come up with something.

**Comment by gujiaxi** [comment]
@joostkremers Great!

**Comment by joostkremers** [comment]
Check out the latest commit: you should now be able to store notes in a single file by setting the option `ebib-notes-use-single-file` to the full path of the notes file. By default, Emacs assumes that this is an org-mode file along the lines of your screen shot, but everything is customisable. See the manual for details: https://joostkremers.github.io/ebib/ebib-manual.html#notes-files

**Comment by joostkremers** [comment]
There are bound to be some rough edges and there may be bugs, so do make regular back ups of your notes file. If anything goes wrong or if you have suggestions for improvements, please let me know.

**Comment by gujiaxi** [comment]
Awesome! It works quite well for now.

**Comment by gujiaxi** [comment]
1. For org-title creation in notes, the title will be in mess if corresponding fields have multilines.
2. btw. Author field may be not that vital so it should be better if org-title is in the following format **by default**.

   ``` lisp
(defun ebib-notes-create-org-title (key)
  (let ((title (or (ebib-db-get-field-value "title" key ebib--cur-db 'noerror 'unbraced 'xref)
                   "(No Title)"))
        (year (or (ebib-db-get-field-value "year" key ebib--cur-db 'noerror 'unbraced 'xref)
                  "????")))
    (format "%s (%s)" title year)))
   ```

**Comment by joostkremers** [comment]
Thanks for your feedback.

> 1. For org-title creation in notes, the title will be in mess if corresponding fields have multilines.

True, hadn't thought of that. I'll have to filter out newline characters.

> 2. btw. Author field may be not that vital so it should be better if org-title is in the following format **by default**.

Actually, from my perspective, the title isn't that important. That may be because I'm in a field where author-date style citations are common, so I'm biased toward that format.

I chose the author-year-title format because I believe it covers the needs / wishes of most users, even if it may be too much for some. Since the title format is customisable, it shouldn't really be a problem. Also, the title is created only once and isn't used by Ebib in any way, so you can change it manually while editing the note and nothing will go wrong.

**Comment by gujiaxi** [comment]
Yes, agree.

**Comment by joostkremers** [comment]
I polished things a bit: the notes buffer is now narrowed to the note you're viewing / creating, which creates a cleaner look IMHO. This can be disabled by customising the hooks `ebib-notes-search-note-before-hook`, `ebib-notes-open-note-after-hook` and `ebib-notes-new-note-hook`.

It may sound a bit overkill to use three hooks where a simple on/off toggle would have sufficed, but this way it's possible to use a format other than org-mode for the notes file. (Not that I expect many people to that, but still.)

I'll close this issue now. If you run into problems or have any further suggestions, I would appreciate it if you could open a new issue rather than add to this one.

Thanks!
